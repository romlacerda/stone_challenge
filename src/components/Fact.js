import React, { Component } from 'react'
import { View, Text } from 'react-native'
import style from '../style/main'

export default props => 
    <View style={style.fact_box}>
        <Text style={style.fontLess80}>{props.value}</Text>
    </View>