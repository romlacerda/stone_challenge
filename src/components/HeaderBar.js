import React, { Component } from 'react'
import { View, Text } from 'react-native'

import styles from '../style/main'
import Icon from 'react-native-vector-icons/MaterialIcons';

export default class HeaderBar extends Component { 
    render() {
        return(
            <View style={styles.header_bar}>
                <Text style={styles.header_title}>{this.props.title}</Text>
                <Icon style={styles.search_icon} name="search" size={30} onPress={() => this.props.navigate('Search')} />
            </View>
        )
    }
    
}