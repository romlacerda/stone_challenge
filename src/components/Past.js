import React, { Component } from 'react'
import { View, Text, Button, TouchableOpacity } from 'react-native'
import style from '../style/main'
import axios from 'axios'
import { connect } from 'react-redux';
import { addFact, searchFact } from '../actions/FactActions';
import { bindActionCreators } from 'redux';
import { Alert } from 'react-native';

class Past extends Component {
    
    ListByFilter = (e, param) => {
        axios.get(`https://api.chucknorris.io/jokes/search`, {
            params: {
                'query': param
            }
        })
        .then(res => {
            this.props.addFact(res.data.result);
            this.props.navigation.navigate('Home')
        }).catch(ex => {
            Alert.alert('Erro', 'Houve um erro de conex√£o.')
            console.log(ex);
        })
    }   

    render() {

        return(
            <View style={style.past_search}>
                <Text style={style.font_default} onPress={(e, param) => this.ListByFilter(e, this.props.past_name)}>{this.props.past_name}</Text>
            </View>
        )
    }
}

const mapStateToProps = (state) => {
    const { facts, searching, searchs } = state
    return { facts, searching, searchs }
  };

const mapDispatchToProps = dispatch => (
    bindActionCreators({
      addFact, searchFact
    }, dispatch)
  );
  
  export default connect(mapStateToProps, mapDispatchToProps)(Past);