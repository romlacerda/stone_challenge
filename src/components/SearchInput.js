import React, { Component } from 'react'
import { View, Text, TextInput } from 'react-native';

import styles from '../style/main';
import axios from 'axios'

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addFact, searchFact } from '../actions/FactActions';
class SearchInput extends Component {

    state = {
        value: null,
    }

    handleChange = (value) => {
        this.setState({ value })
    }

    AddLastSearch = (searched) => {
        let searchs = [...this.state.searchs, this.state.value]
    }

    ListByFilter = (e) => {
        axios.get(`https://api.chucknorris.io/jokes/search`, {
            params: {
                'query': this.state.value
            }
        })
        .then(res => {
            // this.AddLastSearch(this.state.value);
            this.props.addFact(res.data.result);
            this.props.searchFact(this.state.value);
            this.props.navigation.navigate('Home')
        }).catch(ex => {
            console.log(ex);
        })
    }   

    render() {
        return (
            <TextInput placeholder='Digite o que procura...' 
                style={styles.search_input} 
                onChangeText={this.handleChange} 
                onSubmitEditing={(e) => this.ListByFilter(e)}/>
        )
    }
}


const mapStateToProps = (state) => {
    const { facts, searching, searchs } = state
    return { facts, searching, searchs }
  };

const mapDispatchToProps = dispatch => (
    bindActionCreators({
      addFact, searchFact
    }, dispatch)
  );
  
  export default connect(mapStateToProps, mapDispatchToProps)(SearchInput);