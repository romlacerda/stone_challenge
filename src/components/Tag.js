import React, { Component } from 'react'
import { Text, TouchableOpacity } from 'react-native'
import style from '../style/main'
import axios from 'axios'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addFact } from '../actions/FactActions';

class Tag extends Component {

    ListByTag = async (tag) => {
        await axios.get(`https://api.chucknorris.io/jokes/random`, {
            params: {
                'category': tag
            }
        })
        .then(res => {
            // this.props.navigation.navigate('Home', {
            //     tags: res.data
            // })
            const obj = [{
                ...res.data
            }]

            this.props.addFact(obj);
            this.props.navigation.navigate('Home')
        }).catch(ex => {
            alert(error.message);
        })
    } 
 
 
    render() {
        return (
            <TouchableOpacity 
                style={style.tag_button} 
                onPress={() => this.ListByTag(this.props.name)}>
                    <Text style={style.tag_button_font}>{this.props.name}</Text>
            </TouchableOpacity>
        )
    }
}

const mapStateToProps = (state) => {
    const { facts, searching } = state;
    return { facts, searching }
  };

const mapDispatchToProps = dispatch => (
    bindActionCreators({
      addFact, 
    }, dispatch)
  );
  
  export default connect(mapStateToProps, mapDispatchToProps)(Tag);