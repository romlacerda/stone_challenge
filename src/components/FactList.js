import React, { Component } from 'react'
import { View, ScrollView, FlatList } from 'react-native';

import Fact from './Fact';
import style from '../style/main';

import { connect } from 'react-redux';


class Facts extends Component {
    render() {
        
        renderItems = ({item}) => {
            return <Fact {...item} />
        }

        const chuckNorrisFacts = this.props.facts.factList;
        const isSearching = this.props.facts.isSearching;
        return (
            <ScrollView style={{flex: 1}}>
                <View style={style.click_container}>
                {
                    isSearching ? 
                    <FlatList
                        style={style.fact_list}
                        data={chuckNorrisFacts} 
                        renderItem={renderItems} 
                        keyExtractor={(_, index) => index.toString()} 
                    /> : 
                    <View style={style.click_to_search}>
                    </View>
                }
                </View>
            </ScrollView>
        )
    }

}

const mapStateToProps = (state) => {
    const { facts, searching } = state;
    return { facts, searching }
  };

export default connect(mapStateToProps)(Facts);
