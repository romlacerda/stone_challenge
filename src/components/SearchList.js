import React, { Component } from 'react'
import { View, Text } from 'react-native';

import style from '../style/main';
import { connect } from 'react-redux';
import Past from './Past';

export class SearchList extends Component {
    render() {
        return (
            <View style={style.search_list}>
                <Text style={style.search_screen_title}>Ãšltimas pesquisas</Text>
                <View style={style.past_search_container}>
                    {this.props.searchs.searchs.map((data, index) => <Past key={index} navigation={this.props.navigation} past_name={data}/>)}
                </View>
            </View>
        )
    }
}

const mapStateToProps = (state) => {
    const { searchs } = state;
    return { searchs }
  };

export default connect(mapStateToProps)(SearchList);
