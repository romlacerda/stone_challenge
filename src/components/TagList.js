import React, { Component } from 'react'
import { View, Text, FlatList } from 'react-native';

import Tag from './Tag';
import style from '../style/main';

export default class TagList extends Component {
    
    ShuffleTags = () => {
        const shuffled = this.props.tags.sort(() => 0.5 - Math.random());
        let choices = shuffled.slice(0, 8);
        return choices;
    }

    render() {
        return (
            <View style={style.margin_50}>
                <Text style={style.search_screen_title}>Tags</Text>
                <View style={style.tag_search_container}>
                    {/* Não e uma boa pratica colocar o index como key, porém, neste caso aqui, não vi motivo para ser algo diferente. */}
                    {this.ShuffleTags().map((data, index) => <Tag key={index} name={data} navigation={this.props.navigation}/>)}
                </View>
            </View>
        )
    }


}
