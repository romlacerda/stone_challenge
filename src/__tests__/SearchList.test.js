
import React from 'react';
import { shallow, mount } from 'enzyme';
import { SearchList } from '../components/SearchList';
import Past from '../components/Past'
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const initialState = {
    searchs: [{
        searchs: ['star wars', 'the batman', 'hal jordan'] 
    }],
    testo: 'teste'
};

const store = mockStore(initialState);

describe('Testing', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <SearchList searchs={initialState}/>,
            { context: { store }}
        ).debug();

        expect(wrapper).toMatchSnapshot();
    })

});