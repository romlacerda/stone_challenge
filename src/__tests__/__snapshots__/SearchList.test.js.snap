// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing should render correctly 1`] = `
"<View style={{...}}>
  <Text style={{...}}>
    Ãšltimas pesquisas
  </Text>
  <View style={{...}}>
    <Connect(Past) navigation={[undefined]} past_name={{...}} />
  </View>
</View>"
`;
